rpc/internal/system_metrics_service_test.go

```diff
@@ -0,0 +1,334 @@
+		name        string
+		chunks      []*SystemMetricsData
+		env         cedar.Environment
+		invalidConf bool
+		hasErr      bool
+	}{
+		{
+			name: "ValidData",
+			chunks: []*SystemMetricsData{
+				{
+					Id:   systemMetrics.ID,
+					Data: []byte("First byte chunk for valid data"),
+				},
+				{
+					Id:   systemMetrics.ID,
+					Data: []byte("Second byte chunk for valid data"),
+				},
+				{
+					Id:   systemMetrics.ID,
+					Data: []byte("Third byte chunk for valid data"),
+				},
+			},
+			env: env,
+		},
+		{
+			name: "DifferentSystemMetricsIDs",
+			chunks: []*SystemMetricsData{
+				{
+					Id:   systemMetrics.ID,
+					Data: []byte("First byte chunk for different system metrics ids"),
+				},
+				{
+					Id:   systemMetrics2.ID,
+					Data: []byte("Second byte chunk for different system metrics ids"),
+				},
+			},
+			env:    env,
+			hasErr: true,
+		},
+		{
+			name: "LogDNE",
+			chunks: []*SystemMetricsData{
+				{
+					Id:   "DNE",
+					Data: []byte("First byte chunk for invalid system metrics ids"),
+				},
+			},
+			env:    env,
+			hasErr: true,
+		},
+		{
+			name: "InvalidEnv",
+			chunks: []*SystemMetricsData{
+				{
+					Id:   systemMetrics.ID,
+					Data: []byte("First byte chunk for invalid env"),
+				},
+			},
+			env:    nil,
+			hasErr: true,
+		},
+		{
+			name: "InvalidConf",
+			chunks: []*SystemMetricsData{
+				{
+					Id:   systemMetrics.ID,
+					Data: []byte("First byte chunk for invalid conf"),
+				},
+			},
+			env:         env,
+			invalidConf: true,
+			hasErr:      true,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			port := getPort()
+			require.NoError(t, startSystemMetricsService(ctx, test.env, port))
+			client, err := getSystemMetricsGRPCClient(ctx, fmt.Sprintf("localhost:%d", port), []grpc.DialOption{grpc.WithInsecure()})
+			require.NoError(t, err)
+
+			if test.invalidConf {
+				conf.Bucket.SystemMetricsBucket = ""
+			} else {
+				conf.Bucket.SystemMetricsBucket = tempDir
+			}
+			conf.Setup(env)
+			require.NoError(t, conf.Save())
+
+			stream, err := client.StreamSystemMetrics(ctx)
+			require.NoError(t, err)
+
+			catcher := grip.NewBasicCatcher()
+			for i := 0; i < len(test.chunks); i++ {
+				catcher.Add(stream.Send(test.chunks[i]))
+			}
+			resp, err := stream.CloseAndRecv()
+			catcher.Add(err)
+
+			if test.hasErr {
+				assert.Error(t, err)
+				assert.Nil(t, resp)
+			} else {
+				require.NoError(t, err)
+				require.NotNil(t, resp)
+				assert.Equal(t, test.chunks[0].Id, resp.Id)
+
+				sm := &model.SystemMetrics{ID: resp.Id}
+				sm.Setup(env)
+				require.NoError(t, sm.Find(ctx))
+				assert.Equal(t, systemMetrics.ID, systemMetrics.Info.ID())
+				assert.Len(t, sm.Artifact.Chunks, len(test.chunks))
+				for _, key := range sm.Artifact.Chunks {
+					_, err := bucket.Get(ctx, key)
+					assert.NoError(t, err)
+				}
+			}
+		})
+	}
+}
+
+func createSystemMetricsEnv() (cedar.Environment, error) {
+	testDB := "system-metrics-service-test"
```

Comment:
Is there any reason this testDB can't just use the same testDB from other GRPC tests?